src/http/metricshandler.cpp

```diff
@@ -49,8 +56,19 @@ class MetricsHandler::State
         m_session_counters.at(session) = { ms, stats};
     }
 
+    void OnStateUpdate(const std::string& session, const std::vector<libtorrent::torrent_status>& torrents)
+    {
+        for (const auto& status : torrents)
+        {
+            uint64_t ms = duration_cast<std::chrono::milliseconds>(std::chrono::system_clock::now().time_since_epoch()).count();
+            m_torrent_statuses.insert_or_assign(status.info_hashes.v1, std::make_tuple(ms,session, status));
+        }
+    }
+
     boost::signals2::connection m_stats_connection;
+    boost::signals2::connection m_state_update_connection;
     std::map<std::string, std::pair<std::uint64_t, lt::span<const int64_t>>> m_session_counters;
+    std::map<lt::sha1_hash, std::tuple<std::uint64_t, std::string, libtorrent::torrent_status>> m_torrent_statuses;
```

Comment:
Use `lt::info_hash_t` which is the preferred abstraction for an info hash and supports both v1 and v2 hashes.

Reply:
done